doc/DEPLOYMENT.md

```diff
@@ -222,36 +178,47 @@ In the top level of the *jupyterhub-deployment* repository, create a directory s
 - `mkdir -p secrets/deployments/<deployment-name>`
 - `cd secrets/deployments/<deployment-name>`
 
-In the AWS console, find the URL of the secrets repository by navigating to **Services → CodeCommit → Repositories** and click on the repository named *deployment-name-secrets*.  Click on the drop-down button called "Clone URL" and select "Clone HTTPS".  The copied URL should look something like https://git-codecommit.us-east-1.amazonaws.com/v1/repos/deployment-name-secrets.
+In the AWS console, find the URL of the secrets repository by navigating to **Services → CodeCommit → Repositories** and click on the repository named *<deployment-name>-secrets*.  Click on the drop-down button called "Clone URL" and select "Clone HTTPS".  The copied URL should look something like https://git-codecommit.us-east-1.amazonaws.com/v1/repos/deployment-name-secrets.
 
-Next, clone the repository:
+Next, assume the secrets-repo-setup role and clone the repository:
 
+- `aws sts assume-role --role-arn arn:aws:iam::<account-id>:role/<deployment-name>-secrets-repo-setup --role-session-name clone`
+- export AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, and AWS_SESSION_TOKEN with the values returned from the previous command
+- `git config --global credential.helper '!aws codecommit credential-helper $@'`
+- `git config --global credential.UseHttpPath true`
 - `git clone https://git-codecommit.us-east-1.amazonaws.com/v1/repos/<deployment-name>-secrets secrets`
 - `cd secrets`
```

Comment:
is there a way to do this section preceeding with awsudo?

Reply:
Basically, the problem is that you can't use git clone ssh (I don't remember why), so you have to go out through the internet and use git clone https.  That's why I had to use the credential helper.

Reply:
Oh - this is because you have to go through the internet to reach CodeCommit, so only https works.